From d6e7012cf625db6c8c4e2c6b82639a0a060b9718 Mon Sep 17 00:00:00 2001
From: Hemanth Thummala <hemanth.thummala@nutanix.com>
Date: Thu, 14 Apr 2016 13:09:37 -0700
Subject: [PATCH 1018/1096] Mask general purpose signals for notifyd.

Currently there is no signal handling available for notify daemon.
Signals like SIGHUP and SIGUSR1 can lead to terminate the notify
daemon. Masking these signals for notifyd as we are not handling them.

BUG: https://bugzilla.samba.org/show_bug.cgi?id=11840

Signed-off-by: Hemanth Thummala <hemanth.thummala@nutanix.com>
Reviewed-by: Volker Lendecke <vl@samba.org>
Reviewed-by: Stefan Metzmacher <metze@samba.org>

Autobuild-User(master): Stefan Metzmacher <metze@samba.org>
Autobuild-Date(master): Fri Apr 15 15:31:19 CEST 2016 on sn-devel-144

(cherry picked from commit cade673f5fff8a578b8620149688ecc93e981205)
(cherry picked from commit 794d0c2a8cd693c8bc5cba951f320df97974b02f)
---
 source3/smbd/notifyd/notifyd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source3/smbd/notifyd/notifyd.c b/source3/smbd/notifyd/notifyd.c
index 316cdea..49fb9c9 100644
--- a/source3/smbd/notifyd/notifyd.c
+++ b/source3/smbd/notifyd/notifyd.c
@@ -256,6 +256,10 @@ struct tevent_req *notifyd_send(TALLOC_CTX *mem_ctx, struct tevent_context *ev,
 		return tevent_req_post(req, ev);
 	}
 
+	/* Block those signals that we are not handling */
+	BlockSignals(True, SIGHUP);
+	BlockSignals(True, SIGUSR1);
+
 	if (ctdbd_conn == NULL) {
 		/*
 		 * No cluster around, skip the database replication
-- 
2.9.0

